File: LeetCode-30DayChallenge/SingleNumber.java
class Solution {
    public static int singleNumber(int[] nums) {
        int unique = 0;
        for (int i : nums)
            unique ^= i;
        return unique;
    }

    public static void main(String ar[]) {
        System.out.println(singleNumber(new int[] { 4, 1, 2, 1, 2 }));
    }
}

File: LeetCode-30DayChallenge/Go/SingleNumber.go
package main

import "fmt"

func main(){
	nums := [] int{4,1,2,1,2}
	fmt.Println(singleNumber(nums))
}

func singleNumber(nums []int) int {
    var result int
    for i:=0;i<len(nums);i++{
        result ^= nums[i]
    }
    return result
}